From: Jiri Slaby <jslaby@suse.cz>
Date: Mon, 8 Apr 2019 10:54:41 +0200
Subject: Revert "KEYS: restrict /proc/keys by credentials at open time"
Patch-mainline: never, kabi
References: kabi

This reverts commit 52f6c8da89b93b98a1dfc5008018012dc3ffd508, upstream
commit 4aa68e07d845562561f5e73c04aa521376e95252. It depends on commit
4aa68e07d845562561f5e73c04aa521376e95252, but we had to revert that one.

Signed-off-by: Jiri Slaby <jslaby@suse.cz>
---
 security/keys/proc.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/security/keys/proc.c b/security/keys/proc.c
index f2c7e090a66d..ec493ddadd11 100644
--- a/security/keys/proc.c
+++ b/security/keys/proc.c
@@ -187,7 +187,7 @@ static int proc_keys_show(struct seq_file *m, void *v)
 
 	struct keyring_search_context ctx = {
 		.index_key		= key->index_key,
-		.cred			= m->file->f_cred,
+		.cred			= current_cred(),
 		.match_data.cmp		= lookup_user_key_possessed,
 		.match_data.raw_data	= key,
 		.match_data.lookup_type	= KEYRING_SEARCH_LOOKUP_DIRECT,
@@ -207,7 +207,11 @@ static int proc_keys_show(struct seq_file *m, void *v)
 		}
 	}
 
-	/* check whether the current task is allowed to view the key */
+	/* check whether the current task is allowed to view the key (assuming
+	 * non-possession)
+	 * - the caller holds a spinlock, and thus the RCU read lock, making our
+	 *   access to __current_cred() safe
+	 */
 	rc = key_task_permission(key_ref, ctx.cred, KEY_NEED_VIEW);
 	if (rc < 0)
 		return 0;
-- 
2.21.0

